mantidproject/mantid
# Beam Centre Finder Additional Features

Description of work.
The following features were requested as additions

- Continuous plotting of results.

- A verbose mode to keep output workspaces from all iterations.

- An option to set Q limits for just the beam centre finder without switching tab.


**To test:**

- Get the loqdemo dataset from http://download.mantidproject.org/ . There is a sample ISIS data set on the bottom right. 
- Add the loqdemo folder to the directories path in mantid.

- Open the SANS GUI V2, Interfaces-> SANS

- Click on load user file navigate to loqdemo folder and load maskfile.txt

- Click on load batch file, load batch_mode_reduction

- Switch to the beamCentre tab, change the Max iterations to 3, click the verbose box and click run.

- Verify that output graphs are produced. 

- Verify that a workspace group called iteration_# is produced for each iteration containing the output.

- Click on the Initial COM box and verify that the first iteration line output in the log is _Itr 0: (321.735208613, 327.871486798)  SX=15505.9794889  SY=772.794267593_






Fixes #21338. 

**Release Notes** 


---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
